Core/Versioning/KspVersion.cs

```diff
@@ -444,6 +444,54 @@ public static bool TryParse(string input, out KspVersion result)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Searches the build map if it knows the version.
+        /// </summary>
+        /// <returns><c>true</c>, if version is valid, <c>false</c> otherwise.</returns>
+        public bool IsValid()
```

Comment:
This notion of "valid" is specific to the operation you're creating and doesn't apply in other cases. E.g., a version of "1.5" is valid when parsing it from a .ckan file, but this function would return false. This logic should probably live someplace more related to creating fake installs.

Reply:
So in KSPManager?

Reply:
Or change the name to `IsKnown`? This would also clear up the use case.

Reply:
Could you use `IsBuildDefined` instead? I think that's what this check amounts to ultimately.

Reply:
`IsBuildDefined` name is already used:
https://github.com/KSP-CKAN/CKAN/blob/0604a12739acefa30f0ee69223ba43dbb52cbe64/Core/Versioning/KspVersion.cs#L79
Also, it doesn't only check if the build number (if specified) is valid, it works for non-build-defined versions to (i.e. 1.5.1 returns true, 1.5.1.2335 returns true, but 1.5.9 returns false, so does 1.5.1.6666).
What about `InBuildMap` instead?

Reply:
Sorry, I meant check `IsBuildDefined` instead of `IsValid` in `FakeInstance`. In both cases, the check would be true for a known version and false otherwise, right?

Reply:
Something I can't see your new comment on GitHub, but I got it per mail @HebaruSan 

> Sorry, I meant check IsBuildDefined instead of IsValid in FakeInstance. In both cases, the check would be true for a known version and false otherwise, right?

No, IsBuildDefined only checks if there is a build number specified (i.e. not -1), but not if it is the right one. Also IsBuildDefined can't find fake version numbers like 1.4.99.

Reply:
I deleted that comment because I realized it was wrong. But email notifications can't be deleted. :(